keyboards/keebio/bamfk1/keymaps/via/keymap.c

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Danny Nguyen (@nooges)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _MAIN,
+    _FN1,
+    _FN2,
+    _FN3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_MAIN] = LAYOUT(
+        RGB_MOD,            // Big Switch
+        FN_MO13, KC_MUTE,   // Encoder presses
+        KC_DOWN, KC_UP,     // Left encoder turns
+        KC_VOLD, KC_VOLU    // Right encoder turns
+    ),
+    [_FN1] = LAYOUT(
+        KC_B,               // Big Switch
+        _______, KC_C,      // Encoder presses
+        KC_PGDN, KC_PGUP,   // Left encoder turns
+        KC_VOLU, KC_VOLD    // Right encoder turns
+    ),
+    [_FN2] = LAYOUT(
+        _______,            // Big Switch
+        _______, _______,   // Encoder presses
+        _______, _______,   // Left encoder turns
+        _______, _______    // Right encoder turns
+    ),
+    [_FN3] = LAYOUT(
+        _______,            // Big Switch
+        _______, _______,   // Encoder presses
+        _______, _______,   // Left encoder turns
+        _______, _______    // Right encoder turns
+    )
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    uint8_t layer = get_highest_layer(layer_state);
```

Comment:
`keymap_key_to_keycode` would work better, as it automatically handles this.  

but ... at that point, the action_exec hack just works better, overall. 

See worklouder keyboard for reference.

Reply:
Is that all the `encoder_actions.c/h` stuff?

Reply:
yup. 

It's hacky, but because it uses action_exec, it processes everything properly, while tap_code16 will not (basically, quantum codes don't work)

Reply:
Okay, switched things over now to the other hack.